src/VsExtension/NuGetPackage.cs

```diff
@@ -326,8 +326,7 @@ protected override void Initialize()
             ProjectUpgradeHandler = new ProjectUpgradeHandler(this, SolutionManager);
 
             // write default NuGet source to NuGet.Config file
-            var PackageSourceProvider = SourceRepositoryProvider.PackageSourceProvider;
-            PackageSourceProvider.SavePackageSources(PackageSourceProvider.LoadPackageSources());
+            NuGet.Configuration.PackageSourceProvider.MigrateDefaultFeedToNewerProtocol();
```

Comment:
Have you renamed this method on the NuGet3 PR alone? If so, please fix

So, I understand that with this change, you do not save the package sources, during NuGet package initialization. But, only when the sources are changed and saved from the Options dialog. Is that correct?

So, what about when the local nuget.config file is read-only? With recent changes by @zhili1208 to fix https://github.com/NuGet/Home/issues/841, if the local nuget.config file has a clear tag; newly added package sources from the UI are added to the local nuget.config. So that they show up after the addition. If the local nuget.config file is read-only, it will break, right? I would verify the behavior